common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/SwitchBrowserHandler.kt

```diff
@@ -0,0 +1,106 @@
+import com.microsoft.identity.common.logging.Logger
+
+/**
+ * SwitchBrowserHandler is a challenge handler for SwitchBrowserChallenge.
+ * It handles the challenge by selecting a valid browser to launch the Switch browser URI.
+ */
+class SwitchBrowserHandler(
+    private val activity: Activity,
+    private val context: Context,
+    private val customTabsManager: CustomTabsManager,
+    private val browserSelector: IBrowserSelector
+) : IChallengeHandler<SwitchBrowserChallenge, Unit>  {
+
+    companion object {
+        private val TAG = SwitchBrowserHandler::class.simpleName
+    }
+
+    constructor( activity: Activity) : this(
+        activity,
+        activity.applicationContext,
+        CustomTabsManager(activity.applicationContext),
+        AndroidBrowserSelector(activity.applicationContext)
+    )
+
+    /**
+     * Process the SwitchBrowserChallenge, which is a request to switch the browser.
+     * This method will select a valid browser to launch the challenge URI.
+     *
+     * @param switchBrowserChallenge challenge request
+     * @return true if the challenge is handled successfully, false otherwise.
+     */
+    @Throws(ClientException::class)
+    override fun processChallenge(switchBrowserChallenge: SwitchBrowserChallenge) {
+        val methodTag = "$TAG:processChallenge"
+
+        // Select a browser to handle the switch browser challenge
+        val browser = browserSelector.selectBrowser(
+            BrowserDescriptor.getBrowserSafeListForSwitchBrowser(),
+            null
+        )
+        if (browser == null) {
+            val exception = ClientException(
+                ClientException.NO_BROWSERS_AVAILABLE,
+                "No browser found for SwitchBrowserChallenge."
+            )
+            Logger.error(methodTag, "No browser found for SwitchBrowserChallenge.", exception)
+            throw exception
+        }
+
+        // Create an intent to launch the browser
+        val browserIntent: Intent
+        if (browser.isCustomTabsServiceSupported) {
+            Logger.info(methodTag, "CustomTabsService is supported.")
+            //create customTabsIntent
+            if (!customTabsManager.bind(context, browser.packageName)) {
+                Logger.warn(methodTag, "Failed to bind CustomTabsService.")
+                browserIntent = Intent(Intent.ACTION_VIEW)
+            } else {
+                browserIntent = customTabsManager.customTabsIntent.intent
+            }
+        } else {
+            Logger.warn(methodTag, "CustomTabsService is NOT supported")
+            browserIntent = Intent(Intent.ACTION_VIEW)
+        }
+        Logger.info(methodTag, "Launching switch browser request on browser: ${browser.packageName}")
+        browserIntent.setPackage(browser.packageName)
+        browserIntent.setData(switchBrowserChallenge.uri)
+        activity.startActivity(browserIntent)
+    }
+
+    fun unbind() {
+        customTabsManager.unbind()
+    }
```

Comment:
where is this getting called?

Reply:
common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java
public void onDestroy()